sql/core/src/main/scala/org/apache/spark/sql/jdbc/PostgresDialect.scala

```diff
@@ -60,7 +60,12 @@ private object PostgresDialect extends JdbcDialect {
     case "bytea" => Some(BinaryType)
     case "timestamp" | "timestamptz" | "time" | "timetz" => Some(TimestampType)
     case "date" => Some(DateType)
-    case "numeric" | "decimal" => Some(DecimalType.bounded(precision, scale))
+    case "numeric" | "decimal" => if (precision > 0) {
```

Comment:
I would like to confirm just in case; we don't check `scale` in this pr? Probably, this might be related to the discussion: https://github.com/apache/spark/pull/23458#discussion_r245480257

Reply:
Postgres [doc](https://www.postgresql.org/docs/9.6/datatype-numeric.html) says that 

> The precision must be positive, the scale zero or positive. 

What the postgres jdbc driver returned in case of `numeric` was 0 for both scale and precision.
The condition proposed in the linked ticket and currently used [here](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala#L230) was roughly `precision > 0 || scale > 0`, but I can not come up with a valid case having precision <=0 while having scale > 0. 
Is there another case where we would have a decimal with precision 0?
Could someone explain?

Reply:
Yea, but, I think we'd be better to add the check `scale > 0`, too, just for safeguards.

Reply:
I think this is fine actually,. We do not support decimals with precision < 0, so this is most likely enough.

Reply:
Actually, I still agree with @maropu (https://github.com/apache/spark/pull/23456#discussion_r), but it looks okay because this is `PostgresDialect.scala`.

Reply:
if we'd add `|| scale > 0`, we'd allow a percision <= 0,which doesn't make any sense and it is not supported by Spark's decimal. So I think this is fine.